middleware_test.go

```diff
@@ -433,6 +434,100 @@ func TestMethodExclusions(t *testing.T) {
 	newFakeProxy(cfg).RunTests(t, requests)
 }
 
+func TestPreserveURLEncoding(t *testing.T) {
```

Comment:
8 other functions in `middleware_test.go` are also > 60 lines. It's because the test cases are long. I suggest living with it and keeping the tests together rather than splitting them up.